x-pack/packages/ml/trained_models_utils/src/constants/trained_models.ts

```diff
@@ -311,7 +311,7 @@ export type InferenceServiceSettings =
 export type InferenceAPIConfigResponse = {
   // Refers to a deployment id
   inference_id: string;
-  task_type: 'sparse_embedding' | 'text_embedding';
+  task_type: 'sparse_embedding' | 'text_embedding' | 'rerank';
```

Comment:
This type was defined a long time ago when the elasticsearch client didn't contain the type definition. You can use `InferenceTaskType` here instead. 
Or the entire type can be replaced with `InferenceInferenceEndpointInfo` from the elasticsearch client.

Reply:
`type InferenceAPIConfigResponse  = InferenceInferenceEndpointInfo & InferenceServiceSettings`

Reply:
Done